graphql-dgs-codegen-core/src/main/kotlin/com/netflix/graphql/dgs/codegen/generators/java/ClientApiGenerator.kt

```diff
@@ -191,6 +192,22 @@ class ClientApiGenerator(private val config: CodeGenConfig, private val document
         return JavaFile.builder(getPackageName(), javaType.build()).build()
     }
 
+    /**
+     * Generate method name. If there are same method names in type `Query`, `Mutation` and `Subscription`, add suffix.
+     * For example, there are `shows` in `Query`, `Mutation` and `Subscription`, the generated files should be:
+     * `ShowsGraphQLQuery`, `ShowsGraphQLMutation` and `ShowsGraphQLSubscription`
+     */
+    private fun generateMethodName(originalMethodName: String, typeName: String, methodNames: MutableSet<String>): String {
```

Comment:
Thanks for the PR @springmonster !

@srinivasankavitha, wdyt?
I think it will be simpler to understand if we don't keep a state, via `methodNames`, and we agree to be consistent and adopt a naming convention of having as  suffix `*GraphQLSubscription| GraphQLMutation | GraphQLQuery`, depending on the operation. This will be a breaking change, so probably something we could do later.